specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -146,50 +165,144 @@ model TextBlocklist {
-@doc("Block item info in text blocklist.")
-model TextBlockItemInfo {
-  @doc("Block item description.")
-  @maxLength(1024)
+@doc("The request to add blocklistItems to a text blocklist.")
+model AddOrUpdateTextBlocklistItemsOptions {
+  @doc("Array of blocklistItems to add.")
+  blocklistItems: TextBlocklistItem[];
+}
+
+@doc("The response of adding blocklistItems to the text blocklist.")
+model AddOrUpdateTextBlocklistItemsResult {
+  @doc("Array of blocklistItems have been added.")
+  blocklistItems?: TextBlocklistItem[];
+}
+
+@doc("The request to remove blocklistItems from a text blocklist.")
+model RemoveTextBlocklistItemsOptions {
+  @doc("Array of blocklistItemIds to remove.")
+  blocklistItemIds: string[];
+}
+
+@added(ContentSafety.Versions.v2023_10_15_Preview)
+@doc("Project is an entity to manage deployment with specified policy.")
+@resource("projects")
+model Project {
+  @doc("Project name.")
+  @pattern("^[a-zA-Z][0-9A-Za-z-]*[0-9a-zA-Z]$")
+  @key("projectName")
+  @visibility("read", "query")
+  @maxLength(128)
+  @minLength(3)
+  projectName: string;
+
+  @doc("Project description.")
+  @maxLength(255)
   description?: string;
 
-  @doc("Block item content.")
+  @doc("Project modality.")
+  @visibility("read", "query", "create")
+  modality: ContentModality;
+
+  @doc("Project display name.")
   @maxLength(128)
-  text: string;
+  @minLength(3)
+  displayName?: string;
+
+  @doc("Project created time.")
+  @visibility("read", "query")
+  createdAt?: utcDateTime;
+
+  @doc("Project last modified time.")
+  @visibility("read", "query")
+  lastModifiedAt?: utcDateTime;
+
+  @doc("Deployed text policy. This value must be null if the modality is not Text.")
+  textPolicy?: TextPolicy;
+
+  @doc("Deployed image policy.This value must be null if the modality is not Image.")
+  imagePolicy?: ImagePolicy;
+}
+
+@added(ContentSafety.Versions.v2023_10_15_Preview)
+@doc("Modality type.")
+enum ContentModality {
+  @doc("Text modality.")
+  Text,
+  @doc("Image modality.")
+  Image,
+}
+
+@added(ContentSafety.Versions.v2023_10_15_Preview)
+@doc("TextCategoryPolicy. Content which severity less than the acceptLessThan will be accepted.")
+model TextCategoryPolicy {
+  @doc("Text category.")
+  category: TextCategory;
+
+  @doc("Severity less than the threshold will be treated as accepted for the category. The value can only be 2, 4, 6 for FourSeverityLevels. If the threshold is undefined, it means all content are accepted.")
+  acceptSeverityThreshold?: int32;
+}
+
+@added(ContentSafety.Versions.v2023_10_15_Preview)
+@doc("ImageCategoryPolicy. Content which severity less than the acceptLessThan will be accepted.")
+model ImageCategoryPolicy {
+  @doc("Image category.")
+  category: ImageCategory;
+  
+  @doc("Severity less than the threshold will be treated as accepted for the category. The value can only be 2, 4, 6 for FourSeverityLevels. If the threshold is undefined, it means all content are accepted.")
+  acceptSeverityThreshold: int32;
```

Comment:
You say, "If the threshold is undefined...", but this property - unlike line 261 above - is required. You're missing a `?` to make this property optional.

Reply:
updated